### PR TITLE
remove amalgamate form CGManifest

### PR DESCRIPTION
**Pull request by Jinming-Hu**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

closes https://github.com/Azure/azure-sdk-for-cpp/issues/3470#event-6292722039

### CODE REVIEW

**Review by vhvb1989 — COMMENTED** [review]
Sync the dependencies with the Notice.txt file from each package.

I added this entry in the past based on the Notice files. See: https://github.com/Azure/azure-sdk-for-cpp/blob/main/sdk/storage/azure-storage-common/NOTICE.txt#L397

**Comment by Jinming-Hu** [comment]
@vhvb1989 Do you know how to generate `NOTICE.txt` from `cgmanifest.json`?

**Comment by ahsonkhan** [comment]
If you look in component governance in Azure DevOps, you'll see a button that creates the Notice.txt file, based on the updated/checked-in cgmanifest file. However, you might just have to hand modify it, if the generated file doesn't match what it actually should be (or since this cgmanifest change isn't checked-in yet):
https://dev.azure.com/azure-sdk/public/_componentGovernance/186735?_a=components&typeId=10858214&alerts-view-option=active

File: sdk/storage/azure-storage-files-datalake/cgmanifest.json [hunk 1]
```diff
@@ -10,16 +10,6 @@
       },
       "DevelopmentDependency": true
     },
-    {
-      "Component": {
-        "Type": "git",
-        "git": {
-          "RepositoryUrl": "https://github.com/edlund/amalgamate",
```
Comment (ahsonkhan) [inline sdk/storage/azure-storage-files-datalake/cgmanifest.json:17]: Why did we have a dependency on this in the first place?

@vhvb1989 looks like you added it here. Does nlohmann indirectly need it:
https://github.com/Azure/azure-sdk-for-cpp/pull/839/commits/6c24921c429b58ceea4d4d42054604b33b891d8b

**Review by ahsonkhan — APPROVED** [review]

File: sdk/storage/azure-storage-files-datalake/cgmanifest.json (hunk 1 above)
Comment (vhvb1989) [reply sdk/storage/azure-storage-files-datalake/cgmanifest.json:17]: I added based on the current Notice.txt provided by storage.

**Comment by Jinming-Hu** [comment]
Hi @danieljurek , can you please help figure out why the CI is failing?

**Comment by Jinming-Hu** [comment]
ok, thanks @danieljurek . So it' not an issue of the PR itself. Do you think it's safe to ignore the failure and merge this PR anyway?

**Comment by danieljurek** [comment]
> ok, thanks @danieljurek Daniel Jurek FTE . So it' not an issue of the PR itself. Do you think it's safe to ignore the failure and merge this PR anyway?

The underlying issue with `git diff` should be fixed now. I kicked off a [new CI run](https://dev.azure.com/azure-sdk/public/_build/results?buildId=1461249&view=results) and expect it to pass... If it doesn't pass because of the `git diff` problem and the rest of the pipeline passes you're safe to override check enforcer with a `/check-enforcer override` comment.

**Pull request merged**